syntax="proto3";
import "google/protobuf/empty.proto";
option go_package=".;proto";

service User{
  rpc GetUserList(PageInfo) returns (UserListResonse);//用户列表
  rpc GetUserByMobile(MobileRequest) returns(UserInfoResponse);//通过mobile查询用户
  rpc GetUserById(IdRequset) returns(UserInfoResponse);//通过Id查询用户
  rpc CreateUser(CreateUserInfo) returns(UserInfoResponse);//创建用户
  rpc UpdateUser(UpdateUserInfo) returns(google.protobuf.Empty);//更新用户
  rpc CheckPassWord(PassWordCheckInfo) returns(CheckPassWordResponse);//检查密码
}
message PassWordCheckInfo{
  string password          = 1;
  string encryptedPassword = 2;
}
message CheckPassWordResponse{
  bool success = 1;
}
message PageInfo{
  uint32 pn    = 1;
  uint32 pSize = 2;
}
message  MobileRequest{
  string mobile = 1;
}
message IdRequset{
  int32 id = 1;
}
message CreateUserInfo{
  string nickName = 1;
  string passWord = 2;
  string mobile   = 3;
}
message UpdateUserInfo{
  int32 id=1;
  string nickName=2;
  string gender=3;
  uint64 birthDay=4;
}
message UserInfoResponse{
  int32  id       = 1;
  string passWord = 2;
  string mobile   = 3;
  string nickName = 4;
  uint64  birthDay = 5;
  string gender   = 6;
  int32  role     = 7;
}
message  UserListResonse{
  int32                     total = 1;
  repeated UserInfoResponse data  = 2;
}